### PR TITLE
SignedNetworkMap verification fix

### PR DESCRIPTION
**Pull request by kasiastreich**
SignedNetworkMap verification should also include cert path validation,
which was probably moved away by accident, because docs say about the
exception CertPathValidatorException.

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/NetworkMap.kt [hunk 1]
```diff
@@ -63,9 +64,11 @@ class SignedNetworkMap(val raw: SerializedBytes<NetworkMap>, val sig: DigitalSig
      * @throws CertPathValidatorException if the certificate path is invalid.
      * @throws SignatureException if the signature is invalid.
      */
-    @Throws(SignatureException::class)
-    fun verified(): NetworkMap {
+    @Throws(SignatureException::class, CertPathValidatorException::class)
+    fun verified(trustedRoot: X509Certificate): NetworkMap {
         sig.by.publicKey.verify(raw.bytes, sig)
+        // Assume network map cert is issued by the root.
```
Comment (rnicoll) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/NetworkMap.kt]: We're explicitly expecting them not to be issued by the root in https://github.com/corda/enterprise/pull/131 , but I think what you're checking is the network map certificate is in the same hierarchy at the default trust root, which is correct.

Could you rephrase to something like:

// Assume network map cert is under the default trust root.

Comment (kasiastreich) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/NetworkMap.kt]: Addressed
